core/src/main/kotlin/io/saagie/updatarium/model/ChangeSet.kt

```diff
@@ -62,7 +64,7 @@ data class ChangeSet(
     fun execute(configuration: UpdatariumConfiguration = UpdatariumConfiguration()): List<ChangeSetError> {
         val exceptions: MutableList<ChangeSetError> = mutableListOf()
         val persistEngine = configuration.persistEngine
-        if (!persistEngine.notAlreadyExecuted(calculateId())) {
+        if (!shouldExecute(persistEngine)) {
```

Comment:
maybe inverse the call: `persistEngine.shouldExecute(changeLog = this)`
(shameless reason: It 's just help me for a later merge)